dev-docs/RFCs/v5.3/attribute-buffer-rfc.md

```diff
@@ -0,0 +1,36 @@
+# RFC: Attribute Buffer Handling
+
+* **Authors**: Xiaoji Chen
+* **Date**: Mar 21, 2018
+* **Status**: **Pending Review**
+
+Related discussions:
+[#1527](https://github.com/uber/deck.gl/pull/1527)
+
+
+## Background
+
+Since v4, applications are allowed to calculate attributes outside of a layer and then pass in the attribute value as a layer prop. AttributeManager skips the default updating procedure if an external buffer prop is found for an attribute.
+
+Currently, `AttributeManager` requires the external buffer to be a typed array. The typed array is later used to construct a luma.gl Buffer object when calling `model.setAttribute`.
+
+The current approach places some limitations on applications:
+- **GPGPU**: If the app is using transform feedback to calculate an attribute, the attribute value is not accessible to the CPU. It will be more efficient to just use the target Buffer of the transform feedback for the attribute, and avoiding the GPU-CPU-GPU roundtrip.
+- **Partial attribute updates**: If the layer only wants to modify a small part of the attribute value, it is more efficient to call `buffer.subData` than uploading the entire attribute array (e.g. editing with Nebula.gl)
+- **Multi-model**: For layers that reuse attributes cross models, a new buffer is constructed for each attribute and each model.
+- **Resource management**: `model.delete()` does not remove existing buffers as they might be in use by other models. Therefore adding/removing layers leave behind orphan buffers which are no longer used and whose resources are never released.
+
+## Proposal: extend luma.gl's Buffer class
+
+- Add `buffer.clone()` method to support two Buffers instances sharing the same GPU buffer but using different data layouts. (Multi-model use case)
```

Comment:
The current classes are rather religiously one-to-one mappings of the underlying WebGL API. The

Reply:
Cloning also requires reference counting when finalizing. There is some initial support for that in the `Resource` base class, or it can be handled in the `Attribute` class